src/xtopcom/xtopcl/src/api_method.cpp

```diff
@@ -685,8 +690,30 @@ void ApiMethod::query_miner_info(std::string & account, std::ostringstream & out
     string null_out = g_userinfo.account + " account has not registered miner.";
     tackle_null_query(out_str, null_out);
 }
-
-void ApiMethod::register_node(const double & mortgage_d,
+int ApiMethod::parse_top_double(const std::string &amount, const uint32_t unit, uint64_t &out)
```

Comment:
looks like `uint32_t unit` equals TOP_UNIT_LENGTH forever, so can we eliminate this parameter?